qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -99,6 +99,13 @@ def save_page(self):
         """Save the current page."""
         self._widget.triggerPageAction(QWebEnginePage.SavePage)
 
+    def show_source(self, dispatcher, url):
+        # use view-source: scheme to show page source for webengine
+        url = QUrl('view-source:{}'.format(url.toString()))
```

Comment:
Starting with Qt 5.8, there's the [QWebEnginePage.ViewSource](http://doc.qt.io/qt-5/qwebenginepage.html#WebAction-enum) action which should be used instead of opening the URL by hand.

Also, I'm not sure what formatting flags to pass to `toString` here, but I'm pretty sure the default doesn't do the right thing. You might want to check what happens with the WebAction with URLs with username/password, with URLs with UTF-8, and with URLs with special characters in it (like a `?query`).